test/lint.js

```diff
@@ -46,42 +46,45 @@ function checkSchema(dataFilename) {
     console.log('   ' + ajv.errorsText(ajv.errors, {
       separator: '\n    ',
       dataVar: 'item'
-    })
-  );
-}
+    }));
+  }
 }
 
 
 function load() {
-  var dirAbs;
+  for (let file of arguments) {
+    if (file.indexOf(__dirname) !== 0) {
+      file = path.resolve(__dirname, '..', file);
+    }
 
-  function processFilename(fn) {
-    var fp = path.join(dirAbs, fn);
-    // If the given filename is a directory, recursively load it.
-    if (fs.statSync(fp).isDirectory()) {
-      load(fp);
-    } else if (path.extname(fn) === '.json') {
-      console.log(dirRel.replace('../', '') + '/' + fn);
-      checkStyle(fp)
-      checkSchema(fp);
+    if (fs.statSync(file).isFile()) {
+      if (path.extname(file) === '.json') {
+        console.log(file.replace(path.resolve(__dirname, '..') + path.sep, ''));
+        checkStyle(file)
+        checkSchema(file);
+      }
+
+      continue;
     }
-  }
 
-  for (const dir of arguments) {
-    dirRel = path.relative(__dirname, dir);
-    dirAbs = path.resolve(__dirname, '../', dir);
-    fs.readdirSync(dirAbs).forEach(processFilename);
+    load.apply(null, fs.readdirSync(file).map((subfile) => {
```

Comment:
Yes, that would work the same. However, using `apply` + `map` allows to reduce the number of function call to `load` for the same amount of loops.